### PR TITLE
feat: Handle future timestamps in ingestion

### PR DESCRIPTION
**Pull request by tomasfarias**
## Problem

We need more information to identify which events are causing the warning to pop off.

Moreover, as the issue seems to be deep-rooted, we should not let incorrect timestampts propagate into our data stores.

## Changes

Let's add the event UUID and name to the warning to help us identify these events while debugging. However, these changes do not expose the new fields to the UI for the time being, so not frontend changes.

To tackle timestamps in the future, we'll be defaulting to provided `timestamp` fields (if available) or the server ingestion time in the event that the computed timestamp ends up at least a day in the future.

### CODE REVIEW

**Comment by tomasfarias** [comment]
Not stale. Had offsite last week and efforts were concentrated somewhere else 😄

File: plugin-server/src/worker/ingestion/timestamps.ts [hunk 1]
```diff
@@ -34,52 +34,72 @@ export function parseEventTimestamp(data: PluginEvent, callback?: IngestionWarni
...
+    }
+
+    if (data['offset']) {
+        parsedTs = now.minus(Duration.fromMillis(data['offset']))
+    }
+
+    const nowDiff = parsedTs.toUTC().diff(now).toMillis()
+
     // Events in the future would indicate an instrumentation bug, lets' ingest them
     // but publish an integration warning to help diagnose such issues.
```
Comment (tiina303) [inline plugin-server/src/worker/ingestion/timestamps.ts:87]: nit: maybe extend the comment to say use we ignore `sent_at` and just use timestamp

File: plugin-server/src/worker/ingestion/timestamps.ts [hunk 2]
```diff
@@ -34,52 +34,72 @@ export function parseEventTimestamp(data: PluginEvent, callback?: IngestionWarni
...
-    if (now.isValid && parsedTs.toUTC().diff(now).toMillis() > FutureEventHoursCutoffMillis) {
+    if (nowDiff > FutureEventHoursCutoffMillis) {
         callback?.('event_timestamp_in_future', {
             timestamp: data['timestamp'] ?? '',
             sentAt: data['sent_at'] ?? '',
             offset: data['offset'] ?? '',
             now: data['now'],
             result: parsedTs.toISO(),
+            eventUuid: data['uuid'],
+            eventName: data['event'],
```
Comment (tiina303) [inline plugin-server/src/worker/ingestion/timestamps.ts:96]: Extend https://github.com/PostHog/posthog/blob/77fd7ffad6b24de5fc8eab2c355d860d0cd8d06b/frontend/src/scenes/data-management/ingestion-warnings/IngestionWarningsView.tsx#L87 (could be a separate follow-up PR), specifically linking to an event search for this uuid would be super helpful to understand the error/problem better for users.

**Review by tiina303 — APPROVED** [review]
🚢

**Pull request merged**